### PR TITLE
chore: add UpdateUsergroupMembership

### PR DESCRIPTION
**Pull request by carolinaecalderon**
## Description
For better code organization, adding this function to the usergroup module instead of EE oidc module. This method is used in EE oidc/service.go (and maybe saml/service.go).

## Test Plan
added an intg test

## Commentary (optional)
Should I unexport all the Tx functions now that `UpdateUserGroupMembership` is in the same module -- or leave as is?

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: master/internal/usergroup/postgres_groups.go [hunk 1]
```diff
@@ -485,3 +485,54 @@ func UsersInGroupTx(ctx context.Context, idb bun.IDB, gid int) ([]model.User, er
 
 	return users, errors.Wrapf(db.MatchSentinelError(err), "Error getting group %d info", gid)
 }
+
+// UpdateUserGroupMembershipTx takes in slice of groups, and updates a user's membership in those groups.
+func UpdateUserGroupMembershipTx(ctx context.Context, tx bun.IDB, u *model.User, groups []string) error {
```
Comment (NicholasBlaskey) [inline master/internal/usergroup/postgres_groups.go:490]: testing?

File: master/internal/usergroup/postgres_groups.go [hunk 2]
```diff
@@ -485,3 +485,54 @@ func UsersInGroupTx(ctx context.Context, idb bun.IDB, gid int) ([]model.User, er
...
+	// Get a list of groups a user is in.
+	currentGroups, err := SearchGroupsWithoutPersonalGroupsTx(ctx, tx, "", u.ID)
+	if err != nil {
+		return fmt.Errorf("finding current user groups: %w", err)
+	}
+
+	var groupsToRemove []int
```
Comment (NicholasBlaskey) [inline master/internal/usergroup/postgres_groups.go:497]: nit could use ```"github.com/determined-ai/determined/master/pkg/set"```

File: master/internal/usergroup/postgres_groups.go [hunk 3]
```diff
@@ -485,3 +485,54 @@ func UsersInGroupTx(ctx context.Context, idb bun.IDB, gid int) ([]model.User, er
...
+			groupsToRemove = append(groupsToRemove, g.ID)
+		}
+	}
+	if len(groupsToRemove) != 0 {
+		if err := RemoveUsersFromGroupsTx(ctx, tx, groupsToRemove, u.ID); err != nil {
+			return fmt.Errorf("failed to remove user from group: %w", err)
+		}
+	}
+
+	var groupsToAdd []int
```
Comment (NicholasBlaskey) [inline master/internal/usergroup/postgres_groups.go:510]: nit could use "github.com/determined-ai/determined/master/pkg/set"

**Review by NicholasBlaskey — APPROVED** [review]

File: master/internal/usergroup/postgres_groups.go (hunk 2 above)
Comment (carolinaecalderon) [reply master/internal/usergroup/postgres_groups.go]: nice package, thanks for reminding me

File: master/internal/usergroup/postgres_groups.go (hunk 1 above)
Comment (carolinaecalderon) [reply master/internal/usergroup/postgres_groups.go:491]: added a test

**Pull request merged**